libraries/botframework-connector/src/auth/appCredentials.ts

```diff
@@ -170,6 +170,11 @@ export abstract class AppCredentials implements msrest.ServiceClientCredentials
     protected abstract async refreshToken(): Promise<adal.TokenResponse>;
 
     private shouldSetToken(webResource: msrest.WebResource): boolean {
+        // Don't set token if appId is falsy or set to anonymous skill appId
+        if (!this.appId || this.appId === AuthenticationConstants.AnonymousSkillAppId) {
```

Comment:
This seems incorrect.  I do not understand why the c# is doing this also. Maybe I'm missing something.  I thought AppCredentials are only ever created using values from env, settings or config.  Are there cases where an AppCredentials are created based on something incoming, and not settings?  How does this end up as AnonymousSkillAppId?

Reply:
This code was (as far as I recall) a faithful port of the C# code. Perhaps @gabog can weigh in here.

Reply:
I think we've come to (at least a temporary) resolution on this.